flytekit/annotated/python_function_task.py

```diff
@@ -186,6 +204,82 @@ def get_command(self, settings: RegistrationSettings) -> List[str]:
             "{{.rawOutputDataPrefix}}",
         ]
 
+    def compile_into_workflow(
+        self, ctx: FlyteContext, task_function: Callable, **kwargs
+    ) -> Union[_dynamic_job.DynamicJobSpec, _literal_models.LiteralMap]:
+        with ctx.new_compilation_context(prefix="dynamic"):
+            workflow_metadata = WorkflowMetadata(on_failure=WorkflowFailurePolicy.FAIL_IMMEDIATELY)
+            defaults = WorkflowMetadataDefaults(interruptible=False)
+
+            self._wf = Workflow(task_function, metadata=workflow_metadata, default_metadata=defaults)
+            self._wf.compile(**kwargs)
+
+            wf = self._wf
+            sdk_workflow = wf.get_registerable_entity()
```

Comment:
this might impact what @wild-endeavor is doing

Reply:
I will wait and rebase after his change is in

Reply:
talked offline, i'll merge this as is